bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/BitcoindRpcClient.scala

```diff
@@ -112,6 +120,82 @@ class BitcoindRpcClient(val instance: BitcoindInstance)(implicit
 
   override def downloadBlocks(
       blockHashes: Vector[DoubleSha256Digest]): Future[Unit] = FutureUtil.unit
+
+  override def processHeaders(headers: Vector[BlockHeader]): Future[ChainApi] =
+    Future.successful(this)
+
+  override def processFilterHeaders(
+      filterHeaders: Vector[FilterHeader],
+      stopHash: DoubleSha256DigestBE): Future[ChainApi] =
+    Future.successful(this)
+
+  override def getHeader(
+      hash: DoubleSha256DigestBE): Future[Option[BlockHeaderDb]] =
+    getBlockHeader(hash).map(header => Some(header.blockHeaderDb))
+
+  override def getHeadersBetween(
+      from: BlockHeaderDb,
+      to: BlockHeaderDb): Future[Vector[BlockHeaderDb]] = {
+    val headerFs =
+      from.height.to(to.height).map(height => getHeaderAtHeight(height))
+    Future.sequence(headerFs).map(_.toVector)
+  }
+
+  private def getHeaderAtHeight(height: Int): Future[BlockHeaderDb] =
+    for {
+      hash <- getBlockHash(height)
+      header <- getBlockHeader(hash)
+    } yield header.blockHeaderDb
+
+  override def getHeadersAtHeight(height: Int): Future[Vector[BlockHeaderDb]] =
+    getHeaderAtHeight(height).map(header => Vector(header))
+
+  override def getBestBlockHeader(): Future[BlockHeaderDb] =
+    for {
+      hash <- getBestBlockHash
+      header <- getBlockHeader(hash)
+    } yield header.blockHeaderDb
+
+  override def nextBlockHeaderBatchRange(
+      prevStopHash: DoubleSha256DigestBE,
+      batchSize: Int): Future[Option[FilterSyncMarker]] = FutureUtil.none
```

Comment:
Please fail explicitly in these scenarios so it doesn't result in hours of debugging in the future :-). I.e.

```scala
Future.failed(new RuntimeException(s"Bitcoind chainApi doesn't allow you fetch block header batch range"))
```